Corpus Module Cleanup

Olamyy opened the pull request:
Took sometime to rework the corpus module. 
Major changes:

1. Introduced a corpus and corpus path getter.
2. A simple yaml file for keeping tracks of  available datasets.

Olamyy committed:
Intial Documentation Commit

Olamyy committed:
Merge pull request #1 from Olamyy/documentation

Intial Documentation Commit

Olamyy committed:
README.

Olamyy committed:
README.md

Olamyy committed:
Introducd ngram tokenizer

Olamyy committed:
Reworking corpus

Olamyy committed:
Cleaned the corpus module. The loader is now a function. Added tests.

Olamyy committed:
Introduced black for formatting

Olamyy committed:
Removed print statements

Olamyy committed:
Added NIGER_VOLTA_CORPUS folder check.

Olamyy committed:
Fixed failing tests

ruohoruotsi commented:
This one is also broken, let me look at both of them and see if there isn't some meta-pattern we can adopt to facilitate a smoother PR process. I know this is painful for the Travis to be so disagreeable. 

Also we might consider adopting GitHub Actions in the future, as a better CI/CD system

Olamyy commented:
Yeah. They both break at this same point.

I added some txt and yaml files that some functions depend on. However, it doesn't seem like travis copies those when it installs the package. It's a packaging issue. I've added the paths to Manifest.in like you did for the model but that still doesn't work. I've also tried adding the paths to package_dir and data_files in setup but still, the same thing.

ruohoruotsi commented:
no problem, working on it now ... will try some things locally, and save tests via pushes to Travis for when I'm fresher in the AM.
